### PR TITLE
add dafen173 js-dom

### PR DESCRIPTION
**Pull request by dafen173**
### JS-DOM

[Demo](https://dafen173.github.io/js-dom/) |
[Code base](https://github.com/dafen173/frontend-2019-p2p/tree/dom-api-task/submissions/dafen173/js-dom)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/dafen173/js-dom/index.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          <h1 class='headline'>6500 ANSI Lm на улице</h1>
+          <div class="container">
+              Работает проектор типа HITACHI CP-WX8265 3LCD  Lamp  1280х800   WXGA 6500lm  3600уе на экран шириной 3 м.
+          </div>
+          <img class='pic' src='img/first.png' alt="Фото инсталляции"> 
+          <br>
+          <img class='pic2' src='img/0697.png' alt="">           
+        </div>
+    </div>
+
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/index.html]: Remove unused empty lines.

File: submissions/dafen173/js-dom/index.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          </div>
+        </div>  
+        <div class="content">
+          <h1 class='headline'>6500 ANSI Lm на улице</h1>
+          <div class="container">
+              Работает проектор типа HITACHI CP-WX8265 3LCD  Lamp  1280х800   WXGA 6500lm  3600уе на экран шириной 3 м.
+          </div>
+          <img class='pic' src='img/first.png' alt="Фото инсталляции"> 
+          <br>
+          <img class='pic2' src='img/0697.png' alt="">
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/index.html]: Fill `alt` attribute.

File: submissions/dafen173/js-dom/script.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
+
+// Menu button
+const menubtn = document.getElementById('menubtn');
+const dropchild = document.getElementById('dropchild');
+menubtn.addEventListener("click", function(){
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/script.js]: Dropdown doesn't work. You add/remove class `.open` but nothing happening.

File: submissions/dafen173/js-dom/script.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  dropchild.classList.toggle('open');
+});  
+
+
+let inst = [{id:'section1', name:'6500 ANSI Lm на улице', description:'Работает проектор типа HITACHI CP-WX8265 3LCD  Lamp  1280х800   WXGA 6500lm  3600уе на экран шириной 3 м.', img:'img/first.png', img2:'img/0697.png'},
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/script.js]: Try to use a more readable format of the object with data. I mean every key of the object at the new line.

File: submissions/dafen173/js-dom/script.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+			{id:'section2', name:'Сшивка двух ультракороткофокусных проекторов', description:'Работают два проектора типа OPTOMA X320UST	Lamp	1024x768	XGA	4000lm   1395уе (за 1шт). Видеосервер для сшивки изображений порядка 1500уе. С дистанции установки в 66 см получаем проекцию шириной 400 cм.', img:'img/543.png', img2:'img/'},
+			{id:'section3', name:'Спарка двух проекторов по 6500 ANSI Lm', description:'Работает спарка из двух проекторов типа HITACHI CP-WX8265	3LCD	Lamp	1280х800   WXGA	6500lm	   3600уе (за 1шт) на фасад здания. Ширина проекции примерно 6 м.', img:'img/sparka-proj.png', img2:'img/sparka-proj-2.png'},
+			{id:'section4', name:'Видеостена на базе трех проекторов', description:'Видеостена на базе трех проекторов. Проекторы от 1500 уе за шт. Видеосервер порядка 1800уе. Центральная проекция шириной 360см, фланговые проекции по 450 см шириной каждая.', img:'img/videowall-proj.png', img2:'img/videowall-proj2.png'}]
+
+
+
+
+function show (e){
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/script.js]: Use more meaningful and obvious function name. Name of the function should describe her actions.
Also, try to use destruction in the function arguments.

File: submissions/dafen173/js-dom/script.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+	let res = inst.filter( obj => {
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/script.js]: Try to use [Array.prototype.find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) to search for the desired object.

File: submissions/dafen173/js-dom/script.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+	document.querySelector(".container").innerHTML = res[0].description
+	document.querySelector(".pic").src = res[0].img
+	document.querySelector(".pic2").src = res[0].img2
+}
+
+
+
+
+const sect = document.querySelectorAll('.installations')
+sect.forEach( function(elem) {
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/script.js]: Good practice to use [Event Delegation](https://davidwalsh.name/event-delegate).

File: submissions/dafen173/js-dom/script.js [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+	elem.addEventListener('click', show)
+})
+
+
+
+
+
+/*
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/script.js]: Avoid committing commented code - it's bad practice. Use another feature branch for saving code. 
Also, remove unused lines.

File: submissions/dafen173/js-dom/style.css [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+.dropdown-child {
+    background: linear-gradient(to top, #FFFFFF, #696969);
+}
+
+
+
+
+.dropdown-child div {
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/style.css]: Avoid styling by HTML tag.
Try to use HTML tags for markup, class - styles, id - js.

File: submissions/dafen173/js-dom/style.css [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+/*
+.mainmenubtn:hover {
+    background-color: green;
+    }
+*/   
+
+.dropdown {
+    /*position: relative;*/
+    /*display: inline-block;*/
+    float: left;
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/style.css]: This is an old school way for defining blocks position. If browser support allows better to use flex or grid.

File: submissions/dafen173/js-dom/style.css [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+   
+
+    .dropdown {        
+            width: 100%;
+    }
+}
+
+
+
+
```
Comment (A-Ostrovnyy) [inline submissions/dafen173/js-dom/style.css]: Remove commented code(use another feature branch for saving these code) and remove empty 
unused lines.

**Comment by A-Ostrovnyy** [comment]
Try to use one style for defining color names: `#fff, #696969` instead of `white, #696969`
I mean `style.css` line13 and line33 same color but different color names.

File: submissions/dafen173/js-dom/script.js (hunk 3 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/script.js]: I assigned this class only for
@media screen and (max-width: 750px)

so okay?

Comment (dafen173) [reply submissions/dafen173/js-dom/script.js]: ?

Comment (A-Ostrovnyy) [reply submissions/dafen173/js-dom/script.js]: Yes, it's ok.
But mobile viewport starting from 767px and going down up to 320px. So better use this case for @media screen and (max-width: 767px).

Comment (dafen173) [reply submissions/dafen173/js-dom/script.js]: understood, accepted
)

File: submissions/dafen173/js-dom/index.html (hunk 1 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/index.html]: done

https://github.com/dafen173/kottans_frontend/blob/master/task_js_dom/index.html

File: submissions/dafen173/js-dom/index.html (hunk 2 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/index.html]: done

https://github.com/dafen173/kottans_frontend/blob/master/task_js_dom/index.html

File: submissions/dafen173/js-dom/script.js (hunk 3 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/script.js]: done

https://github.com/dafen173/kottans_frontend/blob/master/task_js_dom/style.css

ok?

File: submissions/dafen173/js-dom/index.html (hunk 2 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/index.html]: ok?

File: submissions/dafen173/js-dom/index.html (hunk 1 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/index.html]: ok?

File: submissions/dafen173/js-dom/script.js (hunk 4 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/script.js]: done

https://github.com/dafen173/kottans_frontend/blob/master/task_js_dom/script.js

ok?

File: submissions/dafen173/js-dom/script.js (hunk 5 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/script.js]: done

https://github.com/dafen173/kottans_frontend/blob/master/task_js_dom/script.js

ok?

File: submissions/dafen173/js-dom/script.js (hunk 6 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/script.js]: done

https://github.com/dafen173/kottans_frontend/blob/master/task_js_dom/script.js

ok?

File: submissions/dafen173/js-dom/index.html (hunk 1 above)
Comment (A-Ostrovnyy) [reply submissions/dafen173/js-dom/index.html]: Fix all issues and then push the commit. The current pull request will be updated automatically. And then we review your changes.)

File: submissions/dafen173/js-dom/script.js (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/dafen173/js-dom/script.js]: Use Prettier for automatically formatting your code.

File: submissions/dafen173/js-dom/index.html (hunk 1 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/index.html]: ok, thanks

Comment (dafen173) [reply submissions/dafen173/js-dom/index.html]: > Fix all issues and then push the commit. The current pull request will be updated automatically. And then we review your changes.)

Done. Please, review.

**Comment by A-Ostrovnyy** [comment]
Well done!

File: submissions/dafen173/js-dom/index.html (hunk 1 above)
Comment (dafen173) [reply submissions/dafen173/js-dom/index.html]: Thanks!

**Pull request merged**